Dart/src/com/jetbrains/lang/dart/DartProjectComponent.java

```diff
@@ -58,16 +59,18 @@
   protected DartProjectComponent(@NotNull final Project project) {
     super(project);
 
-    VirtualFileManager.getInstance().addVirtualFileListener(new DartFileListener(project), project);
+    if (!Registry.is("dart.projects.without.pubspec", false)) {
+      VirtualFileManager.getInstance().addVirtualFileListener(new DartFileListener(project), project);
 
-    project.getMessageBus().connect().subscribe(ProjectTopics.PROJECT_ROOTS, new ModuleRootAdapter() {
-      @Override
-      public void rootsChanged(ModuleRootEvent event) {
-        myProjectRootsModificationTracker.incModificationCount();
+      project.getMessageBus().connect().subscribe(ProjectTopics.PROJECT_ROOTS, new ModuleRootAdapter() {
+        @Override
+        public void rootsChanged(ModuleRootEvent event) {
+          myProjectRootsModificationTracker.incModificationCount();
```

Comment:
`myProjectRootsModificationTracker.incModificationCount()` might be useful for internal users as well (if they update SDK without closing the IDE). If you do not mind I'll surround only `DartFileListener.scheduleDartPackageRootsUpdate(myProject);` with this `if`.